src/main/utils/LaunchDarkly.ts

```diff
@@ -0,0 +1,47 @@
+import config from 'config';
+import launchDarkly, { LDClient, LDOptions, LDUser } from 'launchdarkly-node-server-sdk';
+import {FeatureFlagClient} from '../types/FeatureFlagClient';
+
+class LaunchDarkly implements FeatureFlagClient {
+  private static instance: LaunchDarkly
+  private readonly client: LDClient;
+  private readonly ldUser: LDUser;
+
+  private constructor() {
+    const sdkKey: string = config.get('launchDarkly.sdkKey');
+    const options: LDOptions = config.get('featureToggles.enabled') ? { diagnosticOptOut: true } : { offline: true };
+    this.ldUser = { key: 'fact-admin' };
+    this.client = launchDarkly.init(sdkKey, options);
+  }
+
+  public static getInstance(): LaunchDarkly {
+    if (!LaunchDarkly.instance) {
+      LaunchDarkly.instance = new LaunchDarkly();
+    }
+
+    return LaunchDarkly.instance;
+  }
+
+  public async getFlagValue(flag: string, defaultValue: boolean): Promise<boolean> {
+    return this.client.variation(flag, this.ldUser, defaultValue);
+  }
+
+  public async getAllFlagValues(defaultValue: boolean): Promise<{ [p: string]: boolean }> {
+    const flagMap = (await this.client.allFlagsState(this.ldUser)).allValues();
+    for (const key in flagMap) {
+      flagMap[key] = flagMap[key] === null ? defaultValue : flagMap[key];
+    }
+
+    return flagMap;
+  }
+
+  public onFlagChange(callback: Function, defaultValue: boolean, flag?: string) {
```

Comment:
You could possibly make the callback function more strongly-typed. The type would be pretty horrible, though. Something like: `callback: (arg: boolean | { [p: string]: boolean }) => any)`